crates/paimon/src/spec/schema.rs

```diff
@@ -51,3 +51,176 @@ pub struct DataField {
     typ: DataType,
     description: Option<String>,
 }
+
+impl DataField {
+    pub fn new(id: i32, name: String, typ: DataType, description: Option<String>) -> Self {
+        Self {
+            id,
+            name,
+            typ,
+            description,
+        }
+    }
+
+    pub fn id(&self) -> i32 {
+        self.id
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn data_type(&self) -> &DataType {
+        &self.typ
+    }
+
+    pub fn description(&self) -> Option<&String> {
+        self.description.as_ref()
+    }
+
+    pub fn new_id(&self, new_id: i32) -> Self {
+        Self::new(
+            new_id,
+            self.name.clone(),
+            self.typ.clone(),
+            self.description.clone(),
+        )
+    }
+
+    pub fn new_name(&self, new_name: String) -> Self {
+        Self::new(
+            self.id,
+            new_name,
+            self.typ.clone(),
+            self.description.clone(),
+        )
+    }
+
+    pub fn new_description(&self, new_description: String) -> Self {
+        Self::new(
+            self.id,
+            self.name.clone(),
+            self.typ.clone(),
+            Some(new_description),
+        )
+    }
+
+    pub fn as_sql_string(&self) -> String {
```

Comment:
We don't need `format_string` and `as_sql_string`, we can implement `Display` instead.

Reply:
Ok